src/SOS/Strike/strike.cpp

```diff
@@ -1326,6 +1326,26 @@ DECLARE_API(DumpMT)
         }
     }
 
+    ReleaseHolder<ISOSDacInterface8> sos8;
+    if (IsRuntimeVersionAtLeast(7) // Prior to .NET 7, a bug existed that made ALC queries fail.
+        && SUCCEEDED(Status = g_sos->QueryInterface(__uuidof(ISOSDacInterface8), &sos8)))
+    {
+        CLRDATA_ADDRESS assemblyLoadContext = 0;
+        Status = sos8->GetAssemblyLoadContext(TO_CDADDR(dwStartAddr), &assemblyLoadContext);
```

Comment:
The failure for 3.1.28 was in setting Status in these two paths. The command exits with this status and the harness thinks it wailed, even though the status failing is a normal thing since it points out to the default ALC.

Reply:
We should not set Status here if querying for ISOSDacInterface8 fails or if GetAssemblyLoadContext fails and only print the ALC info if they succeed.  You wouldn't need the runtime version check and it would make this command backward compatibly.

Reply:
Ah so this is a leftover failure case. That makes sense. I can change that.